recipes/quickcpplib/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+import os
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd, default_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps
+from conan.tools.files import (
+    get, copy, rmdir, rm, apply_conandata_patches,
+    export_conandata_patches
+)
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.52.0"
+
+
+class QuickcpplibCodeConan(ConanFile):
+    name = "quickcpplib"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ned14/quickcpplib"
+    description = "@ned14: Eliminate all the tedious hassle when making state-of-the-art C++ 17 - 23 libraries!"
+    topics = ("header-only",  "common")
+    package_type = "header-library"
+    settings = "os", "arch", "build_type", "compiler"
+
+    @property
+    def _compiler_required_version(self):
+        return {
+            "gcc": "9",
+            "clang": "10",
+            "Visual Studio": "15",
+            "msvc": "191",
+        }
+
+    @property
+    def _pure_cppstd(self):
+        applied_cppstd = self.settings.compiler.get_safe("cppstd")
+        if not applied_cppstd:
+            applied_cppstd = default_cppstd(self)
+
+        if applied_cppstd:
+            return str(applied_cppstd).replace("gnu", "")
+
+        return "17"
+
+    @property
+    def _needs_span_lite(self):
+        return self._pure_cppstd < "20"
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self._needs_span_lite:
+            self.requires("span-lite/0.10.3")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("quickcpplib does not support macOS.")
```

Comment:
```suggestion
        if self.settings.os == "Macos":
            raise ConanInvalidConfiguration("quickcpplib does not support macOS.")
```
@ngrodzitski could you please point why is not working with Mac? According to the readme it should be allowed: https://github.com/ned14/quickcpplib